daemon/api.go

```diff
@@ -55,7 +55,7 @@ import (
 	"github.com/snapcore/snapd/store"
 )
 
-var api = []*Command{
+var privateAPI = []*Command{
```

Comment:
@kyrofa actually ucred* stuff has exactly a template/hack to do that (attach metadata to things such that once all you have is the remote address from the request/connection you can still find out stuff known at accept time, or listener stuff, that's how canAccess works actually. Probably what we want here is to abstract that a little bit more and then wrap the two listeners in different wrappings and then teach the difference to canAccess. @chipaca or me can help with that if this sounds too nebulous.